### PR TITLE
Auth Middleware

### LINKED ISSUE #128: Create Auth Middleware and Add it to Endpoints that Need it
Or else stephanie will flame us

### PR DESCRIPTION
**Pull request by AndyLiang1**
### What was implemented

12 changed files: 

`AuthMiddleware.js` has the middleware, populates the field `req.userId` on success. 
`TripModel.js` added `isPublic` in anticipation of generateItineraryLink, currently does nothing. 

(10 changed files left) 

**BE Users**
`UserController.js` made it so authentication token now has the user's ID too. 

(9 changed files left) 

**Google Coordinates**
Added error handling, error message concatenation. 

(8 changed files left) 

**BE Trips**
`TripRoute.js` now uses the authMiddleware (line 8) 
Changed all requests to take in `userId` from the `req.userId` field instead. The `req.userId` gets populated from the `AuthMiddleware.js`
Added `next()` so that error is logged too (Note: Didn't do this for every function, only the CRUD's)

`TrpController.js`
Removed all instances of `userId` being returned as part of the response
Added error message concatenation
When editing and deleting, also filters for the `userId`. This way, user1 cannot delete user2's trips, even user1 got "authenticated" using their own auth token. 

(6 changed files left) 

**FE Users**
`thunk.js` merely put accessToken into local storage. I named it `travelersTea_accessToken` to not clash with any other accessTokens that may be in your local storage.
`usersSlice.js` only stored username into userSlice. In future maybe we have more info haha. 

(4 changed files left) 

**FE Trips**
`thunk.js` remove `userId` in params since now we use accessToken.
`service.js` made it so axios sends the request with our token for these 4 routes

`NewTripForm.js` merely removed accessing user's id from the slice, as its not there anymore. 
`SessionController.js` remove `userId` in params since now we use accessToken.

### CODE REVIEW

File: server/controllers/TripController.js [hunk 1]
```diff
@@ -124,9 +124,8 @@ class TripController {
       delete updatedTripDTO.userId
       return updatedTripDTO
     } catch (error) {
-      if (config.server.env === 'DEV')
-        console.error('Error while updating trip:', error)
-      throw new Error('Could not edit trip')
+      error.message = 'Could not create trip: ' + error.message
+      throw error
     }
```
Comment (NLmeng) [inline server/controllers/TripController.js]: reason i put if here is to only provide limited error message to the frontend, any reason to remove it?

**Comment by L0Lmaker** [comment]
Could you add a description for this PR, looks like there are a few different changes apart from the auth middleware. Also dont forget to resolve merge conflicts

File: server/controllers/TripController.js (hunk 1 above)
Comment (AndyLiang1) [reply server/controllers/TripController.js]: Resolved in discord chat!

**Comment by AndyLiang1** [comment]
> Could you add a description for this PR, looks like there are a few different changes apart from the auth middleware. Also dont forget to resolve merge conflicts

Resolved in discord chat!

File: server/middlewares/AuthMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const error = new Error('Unauthorized')
+    error.statusCode = 403
+    throw error
+  }
+  const accessToken = authorizationHeader.substring('Bearer '.length)
+  try {
+    const parsedToken = jwt.verify(accessToken, config.server.jwtSecret)
+
+    if (parsedToken) {
+      req.userId = parsedToken.userId // TODO: see if this is possible
```
Comment (NLmeng) [inline server/middlewares/AuthMiddleware.js:17]: should probably delete this comment?

**Review by NLmeng — APPROVED** [review]

**Pull request merged**